bigstream/transform.py

```diff
@@ -98,7 +102,13 @@ def apply_transform(
 
     # set global number of threads
     ncores = ut.get_number_of_cores()
-    sitk.ProcessObject.SetGlobalDefaultNumberOfThreads(2*ncores)
+    if 'ITK_THREADS' in os.environ and os.environ['ITK_THREADS']:
+        nthreads = min(ncores, int(os.environ["ITK_THREADS"]))
+    elif 'NO_HYPERTHREADING' in os.environ:
+        nthreads = ncores
+    else:
+        nthreads = 2*ncores
+    sitk.ProcessObject.SetGlobalDefaultNumberOfThreads(nthreads)
```

Comment:
The default behavior is to set the number of threads to 2*ncores but there's more flexibility and the user can explicitly use ITK_THREADS

Reply:
Sounds good! This is a code duplication, we could just make this a utility function and call it. I think the other place I saw it was in configure_irm but maybe there is also a function in utility already?